src/lpf2hub.ts

```diff
@@ -439,174 +453,182 @@ export class LPF2Hub extends Hub {
+             * Emits when a motor rotates.
+             * @event LPF2Hub#position
+             * @param {string} port
+             * @param {number} angle the position of the motor related to its absolute zero.
+             */
+            APOS: "position",
+            /**
+             * Emits when a motor starts to rotate.
+             * @event LPF2Hub#load
+             * @param {string} port
+             * @param {number} load Percentage of load of the motor.
+             */
+            LOAD: "load",
+            /**
+             * Emits when an angle (tilt, gyro) sensor is activated.
+             * @event LPF2Hub#angle
+             * @param {string} port
+             * @param {number} x
+             * @param {number} y
+             * @param {number} z (Unavailable from WeDo2 tilt sensor)
+             */
+            ANGLE: "angle",
+            "LPF2-ANGLE": "angle",
+            /**
+             * Emits when a tilt sensor is activated.
+             * @event LPF2Hub#tilt
+             * @param {string} port
+             * @param {number} direction
+             */
+            TILT: "tilt",
+            "LPF2-TILT": "tilt",
+            /**
+             * Emits when a orientation sensor is activated.
+             * @event LPF2Hub#orientation
+             * @param {string} port
+             * @param {number} orientation
+             */
+            ORINT: "orientation",
+            /**
+             * Emits when a tilt sensor is activated.
+             * @event LPF2Hub#impact
+             * @param {string} port
+             * @param {number} direction
+             */
+            IMPCT: "impact",
+            "LPF2-CRASH": "impact",
+            IMP: "impact",
+            /**
+             * Emits when an accelerometer sensor is activated.
+             * @event LPF2Hub#angle
+             * @param {string} port
+             * @param {number} x
+             * @param {number} y
+             * @param {number} z
+             */
+            ACCEL: "acceleration",
+            ROT: "acceleration"
+        };
+
+        if (mode.name === "SPEC_1") {
+            this.emit("color", port.id, values[0]);
+            this.emit("distance", port.id, values[1]);
+            this.emit("reflect", port.id, values[3]);
+            /**
+             * A combined color and distance event, emits when the sensor is activated.
+             * @event LPF2Hub#colorAndDistance
+             * @param {string} port
+             * @param {Color} color Uses LWP3 color codes (0 to 10, -1 is no color).
+             * @param {number} distance Distance indicator, from 10 to 0.
+             */
+            this.emit("colorAndDistance", port.id, values[0], values[1]);
+        } else if (mode.name === "POS" && port.type === Consts.DeviceType.CONTROL_PLUS_TILT) {
+            this.emit("angle", port.id, ...values);
+        } else if (modeToEvent[mode.name]) {
+            this.emit(modeToEvent[mode.name], port.id, ...values);
```

Comment:
This should ideally also normalize the values according to min/max raw and si unit. For example accel in boost hub 1G = 65 while in C+ hub 1G=8000.

Reply:
Do you have any link to the device/firmware mode information files ? I would try to work on a normalization helper to convert values.